src/Manifest.ts

```diff
@@ -1,135 +1,284 @@
+        obj[nodeName] = xmlToJson(item);
+      } else {
+        //@ts-ignore
+        if (typeof obj[nodeName].push == "undefined") {
+          const old = obj[nodeName];
+          obj[nodeName] = [];
+          //@ts-ignore
+          obj[nodeName].push(old);
+        }
+        //@ts-ignore
+        obj[nodeName].push(xmlToJson(item));
+      }
+    }
+  }
+  return obj;
+}
 export interface Metadata {
-    title?: string;
-    author?: string;
-    identifier?: string;
-    language?: string;
-    modified?: string;
+  title?: string;
+  author?: string;
+  identifier?: string;
+  language?: string;
+  modified?: string;
 }
 
 export interface Link {
-    rel?: Array<string>;
-    href?: string;
-    type?: string;
-    title?: string;
-    children?: Array<Link>;
+  rel?: Array<string>;
+  href?: string;
+  type?: string;
+  title?: string;
+  children?: Array<Link>;
 }
 
 export default class Manifest {
-    public readonly metadata: Metadata;
-    public readonly links: Array<Link>;
-    public readonly spine: Array<Link>;
-    public readonly resources: Array<Link>;
-    public readonly toc: Array<Link>;
-    private readonly manifestUrl: URL;
-
-    public static async getManifest(manifestUrl: URL, store?: Store): Promise<Manifest> {
-        const fetchManifest = async (): Promise<Manifest> => {
-            const response = await window.fetch(manifestUrl.href)
-            const manifestJSON = await response.json();
-            if (store) {
-                await store.set("manifest", JSON.stringify(manifestJSON));
-            }
-            return new Manifest(manifestJSON, manifestUrl);
-        };
-
-        const tryToUpdateManifestButIgnoreResult = async (): Promise<void> => {
-            try {
-                await fetchManifest();
-            } catch (err) {
-                // Ignore errors.
-            }
-            return new Promise<void>(resolve => resolve());
-        }
+  public readonly metadata: Metadata;
+  public readonly links: Array<Link>;
+  public readonly spine: Array<Link>;
+  public readonly resources: Array<Link>;
+  public readonly toc: Array<Link>;
+  private readonly manifestUrl: URL;
 
-        // Respond immediately with the manifest from the store, if possible.
-        if (store) {
-            const manifestString = await store.get("manifest");
-            if (manifestString) {
-                // Kick off a fetch to update the store for next time,
-                // but don't await it.
-                tryToUpdateManifestButIgnoreResult();
-                const manifestJSON = JSON.parse(manifestString);
-                return new Manifest(manifestJSON, manifestUrl);
-            }
-        }
+  public static async getManifest(
+    manifestUrl: URL,
+    store?: Store
+  ): Promise<Manifest> {
+    const fetchManifest = async (): Promise<Manifest> => {
+      const isJSONManifest = Boolean(manifestUrl.href.endsWith(".json"));
+
+      const manifestJSON = isJSONManifest
```

Comment:
this could use some comments to make sure we know what is happening?  Is this method only for json manifests?